graphics.ts

```diff
@@ -97,8 +171,159 @@ class Graphics {
+     */
+    drawCircle(x: number, y: number, r:number) {
+        this.ctx.beginPath();
+        this.ctx.arc(x, y, r, 0, 2 * Math.PI);
+        this.ctx.stroke(); 
+    }
+
+    /**
+     * Fills an arc at the given coordinates with the given radius and angles.
+     * @param x The x coordinate for the center of the arc.
+     * @param y The y coordinate for the center of the arc.
+     * @param r The radius for the arc.
+     * @param sa The start angle for the arc, in radians.
+     * @param ea The end angle for the arc, in radians.
+     */
+    fillArc(x: number, y: number, r:number, sa:number, ea: number) {
+        this.ctx.beginPath();
+        this.ctx.arc(x, y, r, sa, ea);
+        this.ctx.fill(); 
+    }
+
+    /**
+     * Draws an arc at the given coordinates with the given radius and angles.
+     * @param x The x coordinate for the center of the arc.
+     * @param y The y coordinate for the center of the arc.
+     * @param r The radius for the arc.
+     * @param sa The start angle for the arc, in radians.
+     * @param ea The end angle for the arc, in radians.
+     */
+    drawArc(x: number, y: number, r:number, sa:number, ea: number) {
+        this.ctx.beginPath();
+        this.ctx.arc(x, y, r, sa, ea);
+        this.ctx.stroke(); 
+    }
+
+    /**
+     * Fills a polygon at the given coorinates with the radius and number of sides specified.
+     * @param x The x coordinate for the center of the polygon.
+     * @param y The y coordinate for the center of the polygon.
+     * @param r The radius of the polygon.
+     * @param n The number of sides in the polygon
+     * @param v The rotation of the polygon in radians. Default is 0.
+     */
+    fillPolygon(x: number, y: number, r: number, n: number, v: number = 0) {
+        this.makePolygon(x, y, r, n, true, v);
+    }
+
+    /**
+     * Draws a polygon at the given coorinates with the radius and number of sides specified.
+     * @param x The x coordinate for the center of the polygon.
+     * @param y The y coordinate for the center of the polygon.
+     * @param r The radius of the polygon.
+     * @param n The number of sides in the polygon
+     * @param v The rotation of the polygon in radians. Default is 0.
+     */
+    drawPolygon(x: number, y: number, r: number, n: number, v: number = 0) {
+        this.makePolygon(x, y, r, n, false, v);
+    }
+
+    /**
+     * The actual implementation of the making of the polygon. Draws the polygon, and fills it if specified.
+     * @param x The x coordinate for the center of the polygon.
+     * @param y The y coordinate for the center of the polygon.
+     * @param r The radius of the polygon.
+     * @param n The number of sides in the polygon
+     * @param fill A boolean indicating if the polygon should be filled.
+     * @param v The rotation of the polygon in radians.
+     */
+    private makePolygon(x: number, y: number, r: number, n: number, fill: boolean, v: number) {

```

Comment:
Great refactoring, creating this method. I think you should refactor it to use already existing methods such as the translate and rotate methods you created earlier :)